src/irr/asset/CMeshManipulator.cpp

```diff
@@ -1551,6 +1551,343 @@ core::smart_refctd_ptr<ICPUBuffer> IMeshManipulator::idxBufferFromTrianglesFanTo
 	return nullptr;
 }
 
+IMeshManipulator::OBB IMeshManipulator::calcOBB_DiTO26(ICPUMeshBuffer* mb)
+{
+    IMeshManipulator::OBB resultOBB;
+
+    //TODO: invalid vertex data handling
+
+    const size_t vtxCnt = mb->calcVertexCount();
+
+    if (vtxCnt == 0u)
+    {
+        _NBL_DEBUG_BREAK_IF(true);
+        return resultOBB;
+    }
+    
+    constexpr uint32_t projCnt = 13;
+    constexpr int np = projCnt * 2;		// Number of points selected along the sample directions
+    core::vectorSIMDf selVtxPos[np];
+    core::vectorSIMDf* minVtxPos = selVtxPos;           // Pointer to first half of selVert where the min points are placed    
+    core::vectorSIMDf* maxVtxPos = selVtxPos + projCnt; // Pointer to the second half of selVert where the max points are placed	
+    std::array<float, projCnt> minProjLen;
+    std::array<float, projCnt> maxProjLen;
+
+    core::vectorSIMDf AABBlen; //axis aligned dimensions of the vertices
+    core::vectorSIMDf AABBmid; //axis aligned mid point of the vertices
+    float AABBArea;        //quality measure of the axis-aligned box 
+
+    core::vectorSIMDf p0, p1, p2; // Vertices of the large base triangle
+    core::vectorSIMDf e0, e1, e2; // Edge vectors of the large base triangle
+    core::vectorSIMDf n;          // Unit normal of the large base triangle 
+    core::vectorSIMDf t0, t1;     // tetrahedron vertices
+
+    core::matrix3x4SIMD b; // The currently best found OBB orientation (transposed)
+    float bestVal;         // the best obb quality value
+    core::vectorSIMDf bMin, bMax, bLen;
+
+    constexpr float eps = 0.000001f;
+
+    const core::vectorSIMDf sampleDir[projCnt] =
+    {
+        core::vectorSIMDf(1.0f, 0.0f, 0.0f),
+        core::vectorSIMDf(0.0f, 1.0f, 0.0f),
+        core::vectorSIMDf(0.0f, 0.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, -1.0f),
+        core::vectorSIMDf(1.0f, -1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, -1.0f, -1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, 0.0f),
+        core::vectorSIMDf(1.0f, -1.0f, 0.0f),
+        core::vectorSIMDf(1.0f, 0.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 0.0f, -1.0f),
+        core::vectorSIMDf(0.0f, 1.0f, 1.0f),
+        core::vectorSIMDf(1.0f, 1.0f, -1.0f)
+    };
+
+    auto constructOBB = [](const core::matrix3x4SIMD& rotation, const core::vectorSIMDf& scale, const core::vectorSIMDf& mid)
```

Comment:
I'd construct from `offset` (local origin) and full extent, then your OBB can have the form
```
column0 = localXaxis
column1 = localYaxis
column2 = localZaxis
column3 = offset 
```

no concatenation of matrices necessary